### PR TITLE
Fix/change default context indicator value in overview

### PR DESCRIPTION
**Pull request by bimal125**
## Changes

* Add format field in map and table query
* Filter by `new_cases_per_million' in table and map query

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] n+1 queries
- [x] flake8 issues
- [x] `print`
- [x] typos
- [x] unwanted comments

## This PR contains valid:

- [ ] tests
- [ ] permission checks (tests here too)

### CODE REVIEW

**Review by Rup-Narayan-Rajbanshi — APPROVED** [review]
looks nice!

**Review by ranjan-stha — APPROVED** [review]

**Review by k9845 — APPROVED** [review]

File: apps/visualization/utils.py [hunk 1]
```diff
@@ -314,17 +326,14 @@ def format_indicator_value(iso3, qs_map):
...
             month=F('indicator_month'),
+            format=F('format'),
         ).order_by('subvariable')
         country_most_recent_qs_iso3_map = {}
         for item in country_most_recent_qs:
             if country_most_recent_qs_iso3_map.get(item['iso3']):
-                country_most_recent_qs_iso3_map[item['iso3']].append(
-                    {'month': item['month'], 'indicator_value': item['indicator_value']}
-                )
+                country_most_recent_qs_iso3_map[item['iso3']].append(format_table_data(item))
```
Comment (thenav56) [inline apps/visualization/utils.py]: Easier to use defaultdict here.

File: apps/visualization/utils.py [hunk 2]
```diff
@@ -340,24 +349,21 @@ def format_indicator_value(iso3, qs_map):
...
             month=F('context_date'),
+            format=F('format'),
         )
         emergency_profile_qs_iso3_map = {}
         for item in emergency_profile_qs:
             if emergency_profile_qs_iso3_map.get(item['iso3']):
-                emergency_profile_qs_iso3_map[item['iso3']].append(
-                    {'month': item['month'], 'indicator_value': item['indicator_value']}
-                )
+                emergency_profile_qs_iso3_map[item['iso3']].append(format_table_data(item))
```
Comment (thenav56) [inline apps/visualization/utils.py]: Easier to use defaultdict here.

File: apps/visualization/utils.py [hunk 3]
```diff
@@ -278,6 +282,13 @@ def get_overview_table_data(
 
     existing_iso3 = Countries.objects.values_list('iso3', flat=True)
 
+    def format_table_data(data):
+        return {
+            'month': data['month'],
+            'indicator_value': data['indicator_value'],
+            'format': data['format'],
+        }
```
Comment (thenav56) [inline apps/visualization/utils.py:290]: Maybe we can directly use OverviewTableDataType instead.

File: apps/visualization/utils.py (hunk 1 above)
Comment (thenav56) [inline apps/visualization/utils.py]: With OverviewTableDataType

```suggestion
                country_most_recent_qs_iso3_map[item['iso3']].append(OverviewTableDataType(**item))
```

**Review by thenav56 — CHANGES_REQUESTED** [review]
Looks good. Few suggestions

File: apps/visualization/utils.py (hunk 3 above)
Comment (bimal125) [reply apps/visualization/utils.py:290]: @navin we have extra attribute 'iso3' in qs, but we don't put iso3 in OverviewTableDataType

File: apps/visualization/utils.py (hunk 1 above)
Comment (bimal125) [reply apps/visualization/utils.py]: @thenav56  we have extra attribute 'iso3' in qs, but we don't put iso3 in OverviewTableDataType

File: apps/visualization/utils.py [hunk 4]
```diff
@@ -314,17 +326,11 @@ def format_indicator_value(iso3, qs_map):
...
         for item in country_most_recent_qs:
-            if country_most_recent_qs_iso3_map.get(item['iso3']):
-                country_most_recent_qs_iso3_map[item['iso3']].append(
-                    {'month': item['month'], 'indicator_value': item['indicator_value']}
-                )
-            else:
-                country_most_recent_qs_iso3_map[item['iso3']] = [
-                    {'month': item['month'], 'indicator_value': item['indicator_value']}
-                ]
+            country_most_recent_qs_iso3_map[item['iso3']] = [format_table_data(item)]
```
Comment (thenav56) [inline apps/visualization/utils.py:333]: We need to append it here right?

File: apps/visualization/utils.py [hunk 5]
```diff
@@ -340,24 +346,18 @@ def format_indicator_value(iso3, qs_map):
...
         for item in emergency_profile_qs:
-            if emergency_profile_qs_iso3_map.get(item['iso3']):
-                emergency_profile_qs_iso3_map[item['iso3']].append(
-                    {'month': item['month'], 'indicator_value': item['indicator_value']}
-                )
-            else:
-                emergency_profile_qs_iso3_map[item['iso3']] = [
-                    {'month': item['month'], 'indicator_value': item['indicator_value']}
-                ]
+            emergency_profile_qs_iso3_map[item['iso3']] = [format_table_data(item)]
```
Comment (thenav56) [inline apps/visualization/utils.py:360]: We need to append it here right?

File: apps/visualization/utils.py [hunk 6]
```diff
@@ -314,17 +326,11 @@ def format_indicator_value(iso3, qs_map):
         ).values('iso3').annotate(
             indicator_value=Max('indicator_value'),
             month=F('indicator_month'),
+            format=F('format'),
         ).order_by('subvariable')
-        country_most_recent_qs_iso3_map = {}
+        country_most_recent_qs_iso3_map = defaultdict()
```
Comment (thenav56) [inline apps/visualization/utils.py:331]:
```suggestion
        country_most_recent_qs_iso3_map = defaultdict(list)
```

File: apps/visualization/utils.py [hunk 7]
```diff
@@ -340,24 +346,18 @@ def format_indicator_value(iso3, qs_map):
...
             context_date__lte=TruncMonth(datetime.today()),
             context_date__gte=TruncMonth(datetime.today() - timedelta(days=365)),
         ).values('iso3').annotate(
             max_indicator_month=Max('context_date'),
             indicator_value=F('context_indicator_value'),
             month=F('context_date'),
+            format=F('format'),
         )
-        emergency_profile_qs_iso3_map = {}
+        emergency_profile_qs_iso3_map = defaultdict()
```
Comment (thenav56) [inline apps/visualization/utils.py:358]:
```suggestion
        emergency_profile_qs_iso3_map = defaultdict(list)
```

File: apps/visualization/utils.py (hunk 3 above)
Comment (thenav56) [reply apps/visualization/utils.py:290]: hmm... okay so we can do this right?
```suggestion
    def format_table_data(data):
        return OverviewTableDataType(
            month=data['month'],
            indicator_value=data['indicator_value'],
            format=data['format'],
        )
```

**Review by thenav56 — CHANGES_REQUESTED** [review]

File: apps/visualization/utils.py [hunk 8]
```diff
@@ -16,6 +16,17 @@
 from utils import get_async_list_from_queryset
 
 
+def get_date_range_for_overview():
+    today = datetime.today()
```
Comment (AdityaKhatri) [inline apps/visualization/utils.py]: Always use django timezone

File: apps/visualization/utils.py (hunk 3 above)
Comment (AdityaKhatri) [reply apps/visualization/utils.py:290]: @bimal125 can you check this?

File: apps/visualization/utils.py [hunk 9]
```diff
@@ -288,23 +310,25 @@ def format_indicator_value(iso3, qs_map):
...
     if region or indicator_id:
         all_filters = {
             'region': region,
             'indicator_id': indicator_id,
             'emergency': emergency,
-            'indicator_month__lte': TruncMonth(datetime.today()),
-            'indicator_month__gte': TruncMonth(datetime.today() - timedelta(days=365)),
+            # NOTE: Apply filter from previous month
+            'indicator_month__lte': TruncMonth(get_date_range_for_overview()['previous_month_date']),
+            'indicator_month__gte': TruncMonth(get_date_range_for_overview()['last_twelve_month_date']),
```
Comment (AdityaKhatri) [inline apps/visualization/utils.py]: Only call get_date_range_for_overview() once

File: apps/visualization/utils.py [hunk 10]
```diff
@@ -340,24 +358,18 @@ def format_indicator_value(iso3, qs_map):
         emergency_profile_qs = CountryEmergencyProfile.objects.filter(
             iso3__in=existing_iso3,
             **filters,
-            context_indicator_id='total_cases',
-            context_date__lte=TruncMonth(datetime.today()),
-            context_date__gte=TruncMonth(datetime.today() - timedelta(days=365)),
+            context_indicator_id='new_cases_per_million',
+            context_date__lte=TruncMonth(get_date_range_for_overview()['previous_month_date']),
+            context_date__gte=TruncMonth(get_date_range_for_overview()['last_twelve_month_date']),
```
Comment (AdityaKhatri) [inline apps/visualization/utils.py]: Only call get_date_range_for_overview() once

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

File: apps/visualization/utils.py (hunk 3 above)
Comment (bimal125) [reply apps/visualization/utils.py:290]: format_table_data is used to ignore duplicated record of month so we should not do this

Comment (bimal125) [reply apps/visualization/utils.py:290]: format_table_data is further used to remove duplicated month so we don't do this change

**Pull request merged**